src/App/adapter/lib/state.js

```diff
@@ -22,37 +22,27 @@ const resolvePath = (label, endpoint) => {
     : null
 }
 
-const enhancePagination = (config) => {
+export function createPagination(config) {
   const { page, pageSize } = config
-  const obj =
-    page && pageSize
-      ? {
-          ...config,
-          limit: pageSize,
-          skip: pageSize * (page - 1),
-        }
-      : config
-  const { page: _, pageSize: __, ...res } = obj
-  return res
-}
 
-export function parseState(state, endpoint) {
-  const config = state.find(({ label }) => label === LABEL_FOR_CONFIG) || {}
+  return pageSize === false
+    ? {}
+    : {
+        limit: pageSize,
+        skip: pageSize * (page - 1),
+      }
+}
 
-  const {
-    label: _,
-    name: __,
-    value: ___,
-    include: included = [],
-    ...base
-  } = enhancePagination(config)
+export function parseState(state, config) {
+  const { endpoint, include: included } = config
```

Comment:
`parseState` no longer has to look for the config object and worry about returning a `base` object -- it receives the config as argument. The pagination object is created in `createPagination`, which is now called directly from `translate`.

Reply:
Not seeing the cost here.

Reply:
It's just less convoluted, in my opinion. `mergeState` has enough to deal with as it is. Generating the `include` and `where` objects should be fully isolated from dealing with the "global" query configuration (`order`, `limit` and `skip`).